pod/perlclass.pod

```diff
@@ -0,0 +1,310 @@
+=head1 NAME
+
+perlclass - Perl class syntax reference
+
+=head1 SYNOPSIS
+
+    use v5.36;
+    use feature 'class';
+
+    class My::Example 1.234 {
+        field $x;
+
+        ADJUST {
+            $x = "Hello, world";
+        }
+
+        method print_message {
+            say $x;
+        }
+    }
+
+    My::Example->new->print_message;
+
+=head1 DESCRIPTION
+
+This document describes the syntax of the Perl's C<class> feature, which
+provides native keywords supporting object-oriented programming paradigm.
+
+=head2 History
+
+Since Perl 5, support for objects resolved around the concept of I<blessing>
+references with a package name. Such reference could then be used to call
+subroutines from the package it was blessed with (or any of its parents). This
+system, while bare-bones, was flexible enough to allow creation of multiple
+more advanced, community-driven systems for object orientation.
+
+Class feature is a core implementation of class syntax which is familiar to
+what one would find in other programming languages. It isn't a C<bless>
+wrapper, but a completely new system built right into the perl interpreter.
+
+=head1 KEYWORDS
+
+Enabling the C<class> feature allows the usage of the following new keywords in
+the scope of current package:
+
+=head2 class
+
+    class NAME BLOCK
+
+    class NAME VERSION BLOCK
+
+    class NAME;
+
+    class NAME VERSION;
+
+The C<class> keyword declares a new package which is intended to be a class.
+All other keywords from the C<class> feature should be used in scope of this
+declaration.
+
+    class WithVersion 1.000 {
+        # class definition goes here
+    }
+
+Classes can be declared in either block or statement syntax. If a block is
+used, the body of the block contains the implementation of the class. If the
+statement form is used, the remainder of the file is used up until the next
+C<class> or C<package> statement.
+
+C<class> and C<package> declarations are similar, but classes automatically get
+a constructor named C<new> - You don't have to (and should not) write one.
+Additionally, in the class BLOCK you are allowed to declare fields and methods.
+
+=head2 field
+
+    field VARIABLE_NAME;
+
+    field VARIABLE_NAME = EXPR;
+
+    field VARIABLE_NAME : ATTRIBUTES;
+
+    field VARIABLE_NAME : ATTRIBUTES = EXPR;
+
+Fields are variables which are visible in the scope of the class - more
+specifically within L</method> and C<ADJUST> blocks. Each class instance get
+their own storage of fields, independent of each other.
+
+A field behaves like a normal lexically scoped variable. It has a sigil and is
+private to the class (though creation of an accessor method will make it
+accessible from the outside). The main difference is that different instances
+access different values in the same scope.
+
+    class WithFields {
+        field $scalar = 42;
+        field @array  = qw(this is just an array);
+        field %hash   = (species => 'Marsian', planet => 'Mars');
```

Comment:
Minor nit: `s/Marsian/Martian/` for the accepted term :)

Reply:
Fixed.